### ISSUE TITLE
[Request or Enhancement] More seamless next episode playback

### ISSUE THREAD

**Issue by mathu204** [Tikipeter/repository.tikipeter#353]
So I am using Fens built in Next up and it works great, the only thing is a lot of the time it doesnt play the next episode right away as Seren does. I think Seren pre loads or caches the next video or does something to make it so when u skip to the end of the video it plays the next one almost automtically without closing the player. As it is right now whether I skip or let it play, the video finishes playing and i am taken back to the home screen or whatever menu i played the video from, and then after waiting about 10-20 seconds the next episode starts up. If Fen could play the next episode more seamlessly it would be great! I believe this is already possible bc when I was using next up the addon, (although with fens next up being much better id rather not) it was able to play the next episode faster and that was even with Fen scraping the next episode right when the current one was done.

EDIT: So the idea would be that if u skipped right to the end without getting the Fen Next Up notficiation, it would immediately start the scrape for the next episode, in the case u let it play and u do get the Next up Notif, i assume Fen does the scrape behind the scenes, so can Fen also precache the beginning of the next video in some form so that when u do skip to the end after the notif, or let it play, instead of backing out and waiting then starting the next video, the next video will almost automatcially start playing

[kodi.log](https://github.com/Tikipeter/repository.tikipeter/files/5793456/kodi.log)

**Comment by Tikipeter** [comment]
I've been thinking this over for a few days. Whilst you are right about Seren's more immediate playback, there are issues with Seren's next up that Fen doesn't suffer from.

Seren uses Kodi playlists to give that more "seamless" next ep playback, whereas Fen let's the episode end and then sends the next one to Kodi for playback, meaning in between there is a re-loading of the previous window. I've sped it up as much as possible (by pre-resolving the next link so it is ready to play), but the gap between playback is the side effect of not using Kodi playlists.

I'm going to look into playlists again, but if Nix can't get them working perfectly in Seren without issues, I doubt I'll be able to.

Long story short..... I'm still thinking about this, but the method used to have more "seamless" playback is possibly more trouble than it's worth. We shall see....

**Comment by mathu204** [comment]
> I've been thinking this over for a few days. Whilst you are right about Seren's more immediate playback, there are issues with Seren's next up that Fen doesn't suffer from.
> 
> Seren uses Kodi playlists to give that more "seamless" next ep playback, whereas Fen let's the episode end and then sends the next one to Kodi for playback, meaning in between there is a re-loading of the previous window. I've sped it up as much as possible (by pre-resolving the next link so it is ready to play), but the gap between playback is the side effect of not using Kodi playlists.
> 
> I'm going to look into playlists again, but if Nix can't get them working perfectly in Seren without issues, I doubt I'll be able to.
> 
> Long story short..... I'm still thinking about this, but the method used to have more "seamless" playback is possibly more trouble than it's worth. We shall see....

Alright thank you for considering it, would you mind telling me what the possible errors are? And I’m not sure if they were fixed in 2.0 or not but Seren’s Next Up has always been fine for me.

**Comment by Junkdewd** [comment]
I think the two of you are doing fine with the way each of you are handling things.  Please let seren fans just go use seren and keep fen working the great way it is. 

We hear enough about fen in the subreddit, having it here is really an insult on your work.

**Comment by mathu204** [comment]
> I think the two of you are doing fine with the way each of you are handling things. Please let seren fans just go use seren and keep fen working the great way it is.
> 
> We hear enough about fen in the subreddit, having it here is really an insult on your work.

The point of this feature request isn’t to insult ones work its to make a request based on the fact that two good addons are existing and there is a feature or something that is, in my opinion done better in another. You don’t have to come and try to twist My words into a bashing post, using your own words if you hear enough about this stuff you can leave the GitHub no? If tiki felt I was disrespecting his work by making this or commenting on it he’s free to tell me that and close it and I wouldn’t ask about it but he didn’t and is considering it. Two great things Can co exist without people like u making it a competition or divisive.

**Comment by Junkdewd** [comment]
I'm not bashing you or your thoughts, I am expressing my opinion on the way I'd like to see Peter continue fen development. 

I think the way he has handled next up is preferential, and I follow his github for my reasons.  I certainly wouldn't have the nerve to tell you to stop following because I disagree with you, as you have done with me.  

I stated what I wanted to the dev, I told you to be more open and not tell me to leave if I don't like your opinion.  I'll have no further comment on the thread, or to you.

**Comment by mathu204** [comment]
> I'm not bashing you or your thoughts, I am expressing my opinion on the way I'd like to see Peter continue fen development.
> 
> I think the way he has handled next up is preferential, and I follow his github for my reasons. I certainly wouldn't have the nerve to tell you to stop following because I disagree with you, as you have done with me.
> 
> I stated what I wanted to the dev, I told you to be more open and not tell me to leave if I don't like your opinion. I'll have no further comment on the thread, or to you.

I didnt tell you to leave I used your comment to you to show you how ridiculous it is to say if u just dont like this one go to the other one, when both can co exist. You can also express your opinon on the way youd like to see development go _without_ saying if you dont like it go to seren. Have a good day.

**Comment by mathu204** [comment]
> > I'm not bashing you or your thoughts, I am expressing my opinion on the way I'd like to see Peter continue fen development.
> > I think the way he has handled next up is preferential, and I follow his github for my reasons. I certainly wouldn't have the nerve to tell you to stop following because I disagree with you, as you have done with me.
> > I stated what I wanted to the dev, I told you to be more open and not tell me to leave if I don't like your opinion. I'll have no further comment on the thread, or to you.
> 
> I didnt tell you to leave I used your comment to you to show you how ridiculous it is to say if u just dont like this one go to the other one, when both can co exist. You can also express your opinon on the way youd like to see development go _without_ saying if you dont like it go to seren. You also didnt tell me to be more open, you did say my thread is an insult to the work hes done which is not the case, you may interpret it that way but that doesnt change what the actual meaning of the post was. Have a good day.

**Comment by streamindemons** [comment]
> I've sped it up as much as possible (by pre-resolving the next link so it is ready to play), but the gap between playback is the side effect of not using Kodi playlists.

@Tikipeter quick question about this part, is this only done for unwatched episdoes? I haven't noticed it with ones i've already watched, was just wondering if i am missing something like I usually do!
